Source/WebCore/loader/CrossOriginAccessControl.cpp

```diff
@@ -122,6 +122,21 @@ ResourceRequest createAccessControlPreflightRequest(const ResourceRequest& reque
             preflightRequest.setHTTPHeaderField(HTTPHeaderName::AccessControlRequestHeaders, headerBuffer.toString());
     }
 
+    if (includeFetchMetadata) {
+        auto requestOrigin = SecurityOrigin::create(request.url());
+        if (requestOrigin->isPotentiallyTrustworthy()) {
+            preflightRequest.setHTTPHeaderField(HTTPHeaderName::SecFetchMode, "cors"_s);
+            preflightRequest.setHTTPHeaderField(HTTPHeaderName::SecFetchDest, "empty"_s);
```

Comment:
Can these values be pulled directly from the request being preflighted? I.e., could something like this work?

```
preflightRequest.setHTTPHeaderField(HTTPHeaderName::SecFetchMode, request.httpHeaderField(HTTPHeaderName::SecFetchMode));
```